packages/system/src/Experimental/Stream/_internal/api.ts

```diff
@@ -0,0 +1,883 @@
+  return M.map_(broadcastedQueuesDynamic_(self, maximumLag), (_) =>
+    pipe(C.managed(_), C.chain(fromQueue()), flattenTake)
+  )
+}
+
+/**
+ * Fan out the stream, producing a dynamic number of streams that have the same elements as this stream.
+ * The driver stream will only ever advance of the `maximumLag` chunks before the
+ * slowest downstream stream.
+ */
+export function broadcastDynamic(maximumLag: number) {
+  return <R, E, A>(self: Stream<R, E, A>) => broadcastDynamic_(self, maximumLag)
+}
+
+/**
+ * Converts the stream to a managed list of queues. Every value will be replicated to every queue with the
+ * slowest queue being allowed to buffer `maximumLag` chunks before the driver is backpressured.
+ *
+ * Queues can unsubscribe from upstream by shutting down.
+ */
+export function broadcastedQueues_<R, E, A>(
+  self: Stream<R, E, A>,
+  n: number,
+  maximumLag: number
+): M.Managed<R, never, A.Chunk<H.HubDequeue<unknown, never, Take.Take<E, A>>>> {
+  return pipe(
+    M.do,
+    M.bind("hub", () => T.toManaged(H.makeBounded<Take.Take<E, A>>(maximumLag))),
+    M.bind("queues", ({ hub }) => M.collectAll(A.fill(n, () => H.subscribe(hub)))),
+    M.tap(({ hub }) => M.fork(runIntoHubManaged_(self, hub))),
+    M.map(({ queues }) => queues)
+  )
+}
+
+/**
+ * Converts the stream to a managed list of queues. Every value will be replicated to every queue with the
+ * slowest queue being allowed to buffer `maximumLag` chunks before the driver is backpressured.
+ *
+ * Queues can unsubscribe from upstream by shutting down.
+ */
+export function broadcastedQueues(n: number, maximumLag: number) {
+  return <R, E, A>(self: Stream<R, E, A>) => broadcastedQueues_(self, n, maximumLag)
+}
+
+/**
+ * Converts the stream to a managed dynamic amount of queues. Every chunk will be replicated to every queue with the
+ * slowest queue being allowed to buffer `maximumLag` chunks before the driver is backpressured.
+ *
+ * Queues can unsubscribe from upstream by shutting down.
+ */
+export function broadcastedQueuesDynamic_<R, E, A>(
+  self: Stream<R, E, A>,
+  maximumLag: number
+): M.Managed<
+  R,
+  never,
+  M.Managed<unknown, never, H.HubDequeue<unknown, never, Take.Take<E, A>>>
+> {
+  return M.map_(toHub_(self, maximumLag), (_) => H.subscribe(_))
+}
+
+/**
+ * Converts the stream to a managed dynamic amount of queues. Every chunk will be replicated to every queue with the
+ * slowest queue being allowed to buffer `maximumLag` chunks before the driver is backpressured.
+ *
+ * Queues can unsubscribe from upstream by shutting down.
+ */
+export function broadcastedQueuesDynamic(maximumLag: number) {
+  return <R, E, A>(self: Stream<R, E, A>) => broadcastedQueuesDynamic_(self, maximumLag)
+}
+
+/**
+ * Allows a faster producer to progress independently of a slower consumer by buffering
+ * up to `capacity` chunks in a queue.
+ *
+ * @note Prefer capacities that are powers of 2 for better performance.
```

Comment:
The pow2 comment doesn't apply to the current impl of queue, we should either take a look if it makes sense to port over the pow2 code in queues too